crates/ir/src/container.rs

```diff
@@ -43,19 +43,25 @@ impl<'c> IrContainerIterator<'c> {
     /// # Safety
     /// Caller must ensure that the data is valid IR in valid version.
     unsafe fn read_instruction(&mut self) -> IrInstruction<'c> {
-        match self.read_op_code() {
+        const ALLOC_REGISTER_NAT32: u16 = IrOpCode::AllocRegisterNat32.into_inner();
+        const DROP_REGISTER: u16 = IrOpCode::DropRegister.into_inner();
+        const LOAD_NAT32: u16 = IrOpCode::LoadNat32.into_inner();
+
+        match self.read_nat16() {
```

Comment:
There's the [`num_enum` crate](https://lib.rs/crates/num_enum) which implements this pattern as a `#[derive]` macro. I think we should be using it since it's a lot more scalable than duplicating all the enum variants here.